sklearn/preprocessing/_data.py

```diff
@@ -976,8 +979,13 @@ def partial_fit(self, X, y=None, sample_weight=None):
             constant_mask = _is_constant_feature(
                 self.var_, self.mean_, self.n_samples_seen_
             )
+            # TODO ugly hack, torch.sqrt does not support torch.float16!
+            var = xp.empty(self.var_.shape, dtype=xp.float64)
```

Comment:
I guess a simple xp.astype(..., xp.float64) could replace this and the following line.

Reply:
I agree. Let's try to do this to see how the diff looks like. The temporary `var` variable should probably be renamed to `var_float64` to make the intention behind the code more explicit.

Reply:
Actually I don't understand why we would not always get `float64` variance values here, based on #12338.